hledger-lib/Text/Megaparsec/Custom.hs

```diff
@@ -244,7 +244,7 @@ customErrorBundlePretty errBundle =
       _ -> Nothing
 
     finds :: (Foldable t) => (a -> Maybe b) -> t a -> Maybe b
-    finds f = asum . map f . toList
+    finds f = getAlt . foldMap (Alt . f)
```

Comment:
I think Alt will be new to many readers. Is it an improvement ?

Reply:
(It was new to me, and I had to go have a discussion on #haskell about it. But I just realised I don't care, disregard this comment.)

Reply:
For a simple one-liner, I actually went back an forth a lot on this one. Ideally I would have used `First` instead of `Alt`, but there are actually two different `First`s: one in `Data.Monoid`, which would be `finds f = getFirst . foldMap (First . f)`, and one in `Data.Semigroup`, which would be `finds f = fmap getFirst . foldMap (fmap First . f)`. The former is in all supported versions of base, but was scheduled to disappear a couple of versions ago, so I don't think it will last much longer. The latter will be around for the forseeable future, but requires a Semigroup instance for Maybe, which isn't in all supported versions. I decided `Alt` was the best choice since it doesn't require messing around with any CPP.